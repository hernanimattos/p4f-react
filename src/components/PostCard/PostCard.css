.post-card {
  padding-left: 1rem;
  position: relative;
  margin-bottom: 2rem;
  border-left: 5px solid;
}
