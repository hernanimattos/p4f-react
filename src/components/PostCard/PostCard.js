import React from 'react';
import PropTypes from 'prop-types';
import './PostCard.css';

const PostCard = (props) => {
  const { title, body } = props || {};
  return (
    <div className="post-card">
      <h2>{title}</h2>
      <p>{body}</p>
    </div>
  );
};

PostCard.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
};

export default PostCard;
