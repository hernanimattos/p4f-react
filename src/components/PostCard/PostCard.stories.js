import React from 'react';
import PostCard from './PostCard';
import { withKnobs, text } from '@storybook/addon-knobs';
// import { action } from '@storybook/addon-actions';

export default {
  title: 'PostCard',
  component: PostCard,
  decorators: [withKnobs],
};

export const normal = () => <PostCard title="titulo" body="kkkk" />;
