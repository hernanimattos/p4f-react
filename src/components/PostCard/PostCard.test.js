import React from 'react';
import { render } from '@testing-library/react';
import PostCard from './PostCard';

test('Snap', () => {
  const PostCP = PostCard.toString();
  expect(PostCP).toMatchSnapshot();
});

test('PostCard class', () => {
  const { container } = render(<PostCard title="nome" />);
  expect(container.firstChild).toHaveClass('post-card');
});
test('PostCard class', () => {
  const { getByText } = render(<PostCard title="nome" />);
  const title = getByText('nome');
  expect(title).toBeInTheDocument();
});
