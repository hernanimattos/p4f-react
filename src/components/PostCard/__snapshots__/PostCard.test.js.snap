// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap 1`] = `
"props => {
  const {
    title,
    body
  } = props || {};
  return /*#__PURE__*/_react.default.createElement(\\"div\\", {
    className: \\"post-card\\",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 8,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(\\"h2\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 7
    }
  }, title), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 10,
      columnNumber: 7
    }
  }, body));
}"
`;
